parts/k8s/kubernetescustomscript.sh

```diff
@@ -496,6 +496,25 @@ configAddons() {
     fi
 }
 
+installAzCli(){
+    AZ_REPO=$(lsb_release -cs)
+    echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | \
+    sudo tee /etc/apt/sources.list.d/azure-cli.list
+    retrycmd_if_failure 3 1 30 curl -L https://packages.microsoft.com/keys/microsoft.asc | sudo apt-key add -
```

Comment:
The two-step approach elsewhere (get artifact / add to key) is more resilient to network flakiness

Reply:
Let's reduce the timeout to 5 seconds and retry, say, 10 times